src/abTestingInfra/permutations.ts

```diff
@@ -0,0 +1,166 @@
+let fs = null;
+
+if (typeof window === "undefined") {
+  fs = require("fs-extra");
+}
+
+import nodePath from "path";
```

Comment:
This looks like a file that is logically separate from React components; closer to the Next.js / page-specific file but not as tightly coupled to Next.js itself. How else might we structure, label the file?

Reply:
TODO AN resume from here

Reply:
See description of new folder structure above